examples_apps/run_gate_synthesis.py

```diff
@@ -0,0 +1,469 @@
+r"""
+Quantum gate synthesis
+======================
+
+This demonstration works through the process used to produce the gate
+synthesis results presented in `“Machine learning method for state
+preparation and gate synthesis on photonic quantum
+computers” <https://arxiv.org/abs/1807.10781>`__.
+
+This tutorial uses the TensorFlow backend of Strawberry Fields, giving us
+access to a number of additional functionalities including: GPU integration,
+automatic gradient computation, built-in optimization algorithms, and other
+machine learning tools.
+
+
+Variational quantum circuits
+----------------------------
+
+A key element of machine learning is optimization. We can use
+TensorFlow's automatic differentiation tools to optimize the parameters
+of variational quantum circuits constructed using Strawberry Fields. In
+this approach, we fix a circuit architecture where the states, gates,
+and/or measurements may have learnable parameters :math:`\vec{\theta}`
+associated with them. We then define a loss function based on the output
+state of this circuit. In this case, we define a loss function such that the
+action of the variational quantum circuit is close to some specified target
+unitary.
+
+.. note::
+
+    For more details on the TensorFlow backend in Strawberry Fields, please see
+    :ref:`machine_learning_tutorial`.
+
+For arbitrary gate synthesis using optimization, we need to make use of
+a quantum circuit with a layer structure that is **universal** - that
+is, by ‘stacking’ the layers, we can guarantee that we can produce *any*
+CV state with at-most polynomial overhead. Therefore, the architecture
+we choose must consist of layers with each layer containing
+parameterized Gaussian *and* non-Gaussian gates. **The non-Gaussian
+gates provide both the nonlinearity and the universality of the model.**
+To this end, we employ the CV quantum neural network architecture
+described below:
+
+.. figure:: https://i.imgur.com/NEsaVIX.png
```

Comment:
That was me 🙂 Reason was because this was originally a standalone Jupyter notebook which I was distributing at a workshop, so relative image paths wouldn't work.